core/terraform/migrate.sh

```diff
@@ -189,6 +189,43 @@ if [ -n "${rp_subnet}" ]; then
   fi
 fi
 
+# DNS Zones migration
+declare -A zones=( \
+  ["privatelink.mysql.database.azure.com"]="module.network.azurerm_private_dns_zone.mysql" \
+  ["privatelink.api.azureml.ms"]="module.network.azurerm_private_dns_zone.azureml" \
+  ["privatelink.cert.api.azureml.ms"]="module.network.azurerm_private_dns_zone.azuremlcert" \
+  ["privatelink.notebooks.azure.net"]="module.network.azurerm_private_dns_zone.notebooks" \
+  ["privatelink.postgres.database.azure.com"]="module.network.azurerm_private_dns_zone.postgres" \
+  ["nexus-${TRE_ID}.${LOCATION}.cloudapp.azure.com"]="module.network.azurerm_private_dns_zone.nexus" \
+  )
+for value in "${!zones[@]}"
+do
+  dns_zone_rid=$(echo "${terraform_show_json}" \
+    | jq -r "select(.values.root_module.child_modules != null) .values.root_module.child_modules[] | select (.address==\"module.network\") | .resources[] | select(.address==\"${zones[$value]}\") | .values.id")
+  if [ -n "${dns_zone_rid}" ]; then
+    terraform state rm "${zones[$value]}"
+    terraform import azurerm_private_dns_zone.non_core[\""$value"\"] "${dns_zone_rid}"
+  fi
+done
+
+# module.network.azurerm_private_dns_zone_virtual_network_link.mysql
```

Comment:
maybe change comment to `import state if module.network.azurerm_private_dns_zone_virtual_network_link.mysql exists`

Reply:
I'll delete it too since every single block works in the same way here.